screens/Navigation/ConversationRequestsListNav.ios.tsx

```diff
@@ -0,0 +1,261 @@
+import { GestureHandlerRootView } from "react-native-gesture-handler";
+import { StackAnimationTypes } from "react-native-screens";
+
+import {
+  NativeStack,
+  navigationAnimation,
+  NavigationParamList,
+} from "./Navigation";
+import ActivityIndicator from "../../components/ActivityIndicator/ActivityIndicator";
+import AndroidBackAction from "../../components/AndroidBackAction";
+import Button from "../../components/Button/Button";
+import ConversationFlashList from "../../components/ConversationFlashList";
+import { showActionSheetWithOptions } from "../../components/StateHandlers/ActionSheetStateHandler";
+import {
+  useChatStore,
+  useCurrentAccount,
+} from "../../data/store/accountsStore";
+import {
+  consentToPeersOnProtocol,
+  sortRequestsBySpamScore,
+  updateConsentStatus,
+} from "../../utils/xmtpRN/conversations";
+
+export default function ConversationRequestsListNav() {
+  const sortedConversationsWithPreview = useChatStore(
+    (s) => s.sortedConversationsWithPreview
+  );
+  const colorScheme = useColorScheme();
+  const account = useCurrentAccount() as string;
+  const navRef = useRef<any>();
+  const [clearingAll, setClearingAll] = useState(false);
+
+  const [selectedSegment, setSelectedSegment] = useState(0);
+  const allRequests = sortedConversationsWithPreview.conversationsRequests;
+  const { likelySpam, likelyNotSpam } = sortRequestsBySpamScore(allRequests);
+  const styles = useStyles();
+
+  const clearAllSpam = useCallback(() => {
+    const methods = {
+      "Clear all!": async () => {
+        setClearingAll(true);
+        // @todo => handle groups here
+        const peers = Array.from(
+          new Set(allRequests.map((c) => c.peerAddress))
+        ).filter((peer) => !!peer) as string[];
+        await consentToPeersOnProtocol(account, peers, "deny");
+        await updateConsentStatus(account);
+        setClearingAll(false);
+        navRef.current?.goBack();
+      },
+      Cancel: () => {},
+    };
+
+    const options = Object.keys(methods);
+
+    showActionSheetWithOptions(
+      {
+        options,
+        destructiveButtonIndex: options.indexOf("Clear all"),
+        cancelButtonIndex: options.indexOf("Cancel"),
+        title:
+          "Do you confirm? This will block all accounts that are currently tagged as requests.",
+        ...actionSheetColors(colorScheme),
+      },
+      (selectedIndex?: number) => {
+        if (selectedIndex === undefined) return;
+        const method = (methods as any)[options[selectedIndex]];
+        if (method) {
+          method();
+        }
+      }
+    );
+  }, [account, colorScheme, allRequests]);
+
+  const navigationOptions = useCallback(
+    ({
+      navigation,
+    }: {
+      route: RouteProp<NavigationParamList, "ChatsRequests">;
+      navigation: any;
+    }) => ({
+      animation: navigationAnimation as StackAnimationTypes,
+      headerTitle: clearingAll
+        ? Platform.OS === "ios"
```

Comment:
This is in a .ios file so no need to check these things

Reply:
Thanks, this is fixed!